server/src/test/java/timely/test/integration/http/HttpApiIT.java

```diff
@@ -403,80 +432,112 @@ public void testQueryWithTagWildcard2() throws Exception {
+                    "sys.cpu.user " + (TEST_TIME + 1L) + " 1.0 tag3=value3 rack=r2",
+                    "sys.cpu.idle " + (TEST_TIME + 2L) + " 2.0 tag3=value3 tag4=value4 rack=r1",
+                    "sys.cpu.idle " + (TEST_TIME + 1000L) + " 3.0 tag3=value3 tag4=value4 rack=r2",
+                    "sys.cpu.idle " + (TEST_TIME + 2000L) + " 4.0 tag3=value3 tag4=value4 rack=r3");
             // Latency in TestConfiguration is 2s, wait for it
-            sleepUninterruptibly(4, TimeUnit.SECONDS);
+            sleepUninterruptibly(TestConfiguration.WAIT_SECONDS, TimeUnit.SECONDS);
             QueryRequest request = new QueryRequest();
+
             request.setStart(TEST_TIME);
             request.setEnd(TEST_TIME + 6000);
+            request.setMsResolution(true);
             SubQuery subQuery = new SubQuery();
             subQuery.setMetric("sys.cpu.idle");
             subQuery.setTags(Collections.singletonMap("rack", "r1|r2"));
             subQuery.setDownsample(Optional.of("1s-max"));
             request.addQuery(subQuery);
-            List<QueryResponse> response = query("https://127.0.0.1:54322/api/query", request);
-            assertEquals(2, response.size());
-            QueryResponse response1 = response.get(0);
-            Map<String, String> tags = response1.getTags();
-            assertEquals(1, tags.size());
-            assertTrue(tags.containsKey("rack"));
-            assertTrue(tags.get("rack").equals("r2"));
-            Map<String, Object> dps = response1.getDps();
-            assertEquals(1, dps.size());
-            Iterator<Entry<String, Object>> entries = dps.entrySet().iterator();
-            Entry<String, Object> entry = entries.next();
-            assertEquals(Long.toString((TEST_TIME / 1000) + 1), entry.getKey());
-            assertEquals(3.0, entry.getValue());
-
-            QueryResponse response2 = response.get(1);
-            Map<String, String> tags2 = response2.getTags();
-            assertEquals(1, tags2.size());
-            assertTrue(tags2.containsKey("rack"));
-            assertTrue(tags2.get("rack").equals("r1"));
-            Map<String, Object> dps2 = response2.getDps();
-            assertEquals(1, dps2.size());
-            Iterator<Entry<String, Object>> entries2 = dps2.entrySet().iterator();
-            Entry<String, Object> entry2 = entries2.next();
-            assertEquals(Long.toString((TEST_TIME / 1000)), entry2.getKey());
-            assertEquals(1.0, entry2.getValue());
+            List<QueryResponse> responses = query("https://127.0.0.1:54322/api/query", request);
+
+            assertEquals("Expected 2 responses for sys.cpu.idle rack=r1|r2", 2, responses.size());
+
+            for(QueryResponse response: responses){
+                assertTrue("Found incorrect metric", "sys.cpu.idle".equals(response.getMetric()));
+                response.getTags().forEach((tagk,tagv) -> {
+                   switch (tagk){
```

Comment:
with this approach, you are not ensuring that all of the expected tags are actually seen. You are only checking it if its returned.